### PR TITLE
fix(github): detect disabled issues during initRepo

### LINKED ISSUE #12408: Use GitHub: hasIssuesEnabled
### What would you like Renovate to be able to do?

Detect disabled issues during initRepo

### If you have any ideas on how this should be implemented, please tell us here.

Detect disabled issues during initRepo

### Is this a feature you are interested in implementing yourself?

Yes

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Detects disabled issues during initRepo

## Context:

Closes #12408

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/platform/github/index.ts [hunk 1]
```diff
@@ -985,6 +986,10 @@ export async function setBranchStatus({
 
 /* istanbul ignore next */
 async function getIssues(): Promise<Issue[]> {
+  // istanbul ignore if
```
Comment (viceice) [inline lib/platform/github/index.ts:989]: Isn't this function fully ignored? See comment before declaration. 🤔

Did you checked if we need all those ignores?

**Review by viceice — COMMENTED** [review]
Just a small coverage question, otherwise LGTM

File: lib/platform/github/index.ts (hunk 1 above)
Comment (rarkins) [inline lib/platform/github/index.ts:989]:
```suggestion
```

Comment (rarkins) [reply lib/platform/github/index.ts]: I attempted to get the initRepo mocking right and gave up..

File: lib/platform/github/index.ts [hunk 2]
```diff
@@ -1064,6 +1072,12 @@ export async function ensureIssue({
   shouldReOpen = true,
 }: EnsureIssueConfig): Promise<EnsureIssueResult | null> {
   logger.debug(`ensureIssue(${title})`);
+  // istanbul ignore if
+  if (config.hasIssuesEnabled === false) {
+    logger.info(
+      'Cannot ensure issue because issues are disabled in this repository'
+    );
```
Comment (zharinov) [inline lib/platform/github/index.ts:1079]:
```suggestion
    );
    return null;
```

File: lib/platform/github/index.ts [hunk 3]
```diff
@@ -1151,6 +1165,12 @@ export async function ensureIssue({
 
 export async function ensureIssueClosing(title: string): Promise<void> {
   logger.trace(`ensureIssueClosing(${title})`);
+  // istanbul ignore if
+  if (!config.hasIssuesEnabled) {
+    logger.info(
+      'Cannot ensure issue because issues are disabled in this repository'
+    );
```
Comment (zharinov) [inline lib/platform/github/index.ts:1172]:
```suggestion
    );
    return;
```

File: lib/platform/github/index.ts [hunk 4]
```diff
@@ -985,6 +986,9 @@ export async function setBranchStatus({
 
 /* istanbul ignore next */
 async function getIssues(): Promise<Issue[]> {
+  if (config.hasIssuesEnabled === false) {
+    return [];
+  }
```
Comment (viceice) [inline lib/platform/github/index.ts:991]: I think we don't need this, as it's never called when issues are disabled.

Comment (rarkins) [inline lib/platform/github/index.ts:991]:
```suggestion
```

**Pull request merged**